content/en/docs/tasks/configmap-secret/managing-secret-using-kubectl.md

```diff
@@ -15,64 +19,66 @@ description: Creating Secret objects using kubectl command line.
 
 ## Create a Secret
 
-A `Secret` can contain user credentials required by pods to access a database.
-For example, a database connection string consists of a username and password.
-You can store the username in a file `./username.txt` and the password in a
-file `./password.txt` on your local machine.
+A `Secret` object stores sensitive data such as credentials
+used by Pods to access services. For example, you might need a Secret to store
+the username and password needed to access a database.
 
-```shell
-echo -n 'admin' > ./username.txt
-echo -n '1f2d1e2e67df' > ./password.txt
-```
-In these commands, the `-n` flag ensures that the generated files do not have
-an extra newline character at the end of the text. This is important because
-when `kubectl` reads a file and encodes the content into a base64 string, the
-extra newline character gets encoded too.
+You can create the Secret by passing the raw data in the command, or by storing
+the credentials in files that you pass in the command. The following commands
+create a Secret that stores the username `admin` and the password `VLc5@#tym$HzBn@8`.
+
+{{< tabs name="Secret data" >}}
+{{% tab name="Raw data" %}}
```

Comment:
I'm not a big fan of tabs. It is a fancy thing and people may argue that tabs can shorten the length of a page (visually), or tabs can help users to focus on the content they really care. I'm more inclined to structuring a page into sections and subsections. If a page has become too long (e.g. >1000lines), we may want to split something out for ease of maintenance.

Tabs are not generic, it is hugo-specific. Their value is very limited and there might be some issues when Hugo is handling it. I believe I have seen some issues when code snippets or some other shortcodes not working inside a tab. We have to consider how a page will be rendered when printed to PDF for reference, for example.

What I'm really suggesting is that, we may want to avoid using shortcodes unless they really bring clear value.

Reply:
If that's the approach we want we should definitely add that as a note to our shortcode guide in the contributor docs. 

What do you think is the more likely use case, from files or literals? We should keep the full steps for one and mention the other as an alternative (that'll keep the page short while focusing on the more common journey)